x-pack/plugins/stack_connectors/server/connector_types/security/tines/render.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { set } from 'lodash/fp';
+import { ExecutorParams } from '@kbn/actions-plugin/server/sub_action_framework/types';
+import { RenderParameterTemplates } from '@kbn/actions-plugin/server/types';
+import { SUB_ACTION } from '../../../../common/connector_types/security/tines/constants';
+
+interface Context {
+  alerts: Array<Record<string, unknown>>;
+}
+
+export const renderParameterTemplates: RenderParameterTemplates<ExecutorParams> = (
+  params,
+  variables
+) => {
+  if (params?.subAction !== SUB_ACTION.RUN) return params;
+
+  // Remove the "kibana" entry from all alerts to reduce weight, the same data can be found in other parts of the alert object.
+  const alerts = (variables.context as Context).alerts.map(({ kibana, ...alert }) => alert);
+
+  let body: string;
+  try {
+    body = JSON.stringify(set('context.alerts', alerts, variables));
```

Comment:
Good point, the standard lodash `set` function does mutate the target, but we use the `set` from `lodash/fp` here, which does not mutate the target, it creates an entirely new `variables` object when it is called. I used it instead of doing the deep clone manually.